api/pkg/s3/bucketlifecycleget.go

```diff
@@ -0,0 +1,81 @@
+// Copyright 2019 The OpenSDS Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package s3
+
+import (
+	"github.com/emicklei/go-restful"
+	"github.com/micro/go-log"
+	"github.com/opensds/multi-cloud/api/pkg/policy"
+	"github.com/opensds/multi-cloud/s3/pkg/model"
+	s3 "github.com/opensds/multi-cloud/s3/proto"
+	"golang.org/x/net/context"
+)
+
+func (s *APIService) BucketLifecycleGet(request *restful.Request, response *restful.Response) {
+	if !policy.Authorize(request, response, "bucket:get") {
+		return
+	}
+	bucketName := request.PathParameter("bucketName")
+	log.Logf("Received request for bucket details in GET lifecycle: %s", bucketName)
+
+	ctx := context.Background()
+	bucket, _ := s.s3Client.GetBucket(ctx, &s3.Bucket{Name: bucketName})
+
+	// convert back to xml struct
+	getLifecycleConf := model.LifecycleConfiguration{}
+	getLifecycleConf.Rule = make([]model.Rule, 0)
+	// convert lifecycle rule to xml Rule
+	for _, lcRule := range bucket.LifecycleConfiguration {
+		xmlRule := model.Rule{}
+
+		xmlRule.Status = lcRule.Status
+		xmlRule.ID = lcRule.ID
+		xmlRule.Filter = converts3FilterToRuleFilter(lcRule.Filter)
+		xmlRule.AbortIncompleteMultipartUpload = converts3UploadToRuleUpload(lcRule.AbortIncompleteMultipartUpload)
+		xmlRule.Transition = make([]model.Transition, 0)
+		for _, transition := range lcRule.Actions {
+			xmlTransition := model.Transition{}
+			xmlTransition.Days = transition.Days
+			//xmlTransition.StorageClass = transition.Tier
+			xmlTransition.Backend = transition.Backend
+			xmlRule.Transition = append(xmlRule.Transition, xmlTransition)
+		}
+		for _, expiration := range lcRule.Actions {
```

Comment:
Need to check if the rule's name is expiration, only expiration rule can be add to xmlRule.Expiration..

Reply:
Resolved in latest commit for GET lifecycle API